### PR TITLE
Deliverable 2

### LINKED ISSUE #28: Skriftelig del av Oblig 2
https://docs.google.com/document/d/16NhP5iyGJbDnktFE-MATkEzknzCj9KrfcqHHwuppHkM/edit

### PR DESCRIPTION
**Pull request by bigmantobs**
Assignment 2 written portion in deliverables
-- Vi må framdeles legge til klassediagramm
Please leave comments for suggested changes and I will edit and make a new pull request once all changes have been implemented.

### CODE REVIEW

File: deliverables/ObligatoriskOppgave2.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+## Transcripts
+
+All transcripts are listed by date at our wiki.
+* Transcript Thursday 13.02
+  * https://github.com/inf112-v20/biblioteket/wiki/Transcript-13.02
+* Transcript Thursday 20.02 
+  * https://github.com/inf112-v20/biblioteket/wiki/Transcripts-20.02
+* Transcript Tuesday 25.02
+  * https://github.com/inf112-v20/biblioteket/wiki/Transcript-25.02
```
Comment (tor-kristian) [inline deliverables/ObligatoriskOppgave2.md]: ## Transcripts

All transcripts are listed by date at our wiki.

* [Transcripts Thursday 13.02](https://github.com/inf112-v20/biblioteket/wiki/Transcript-13.02)
* [Transcripts Thursday 20.02](https://github.com/inf112-v20/biblioteket/wiki/Transcript-20.02)
* [Transcripts Tuesday 25.02](https://github.com/inf112-v20/biblioteket/wiki/Transcript-25.02)
* [Transcripts Thursday 27.02](https://github.com/inf112-v20/biblioteket/wiki/Transcript-27.02)

File: deliverables/ObligatoriskOppgave2.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Kravlista er lang, men det er ikke nødvendig å levere på alle kravene hvis det ikke er realistisk. Det er viktigere at de oppgavene som er utført holder høy kvalitet. Utførte oppgaver skal være ferdige.
+
+
+
+How did the team roles function?
+* **First compulsory → roles worked**
+* **Second compulsory → roles changed**
+  * The team roles had to be swapped for this assignment, but this is to be expected, as the further our development 
+goes, the necessity of different roles will shift as the requirements become more clear.
```
Comment (tor-kristian) [inline deliverables/ObligatoriskOppgave2.md]: Flytt opp til "Roles"

File: deliverables/ObligatoriskOppgave2.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Prior to the release of the second assignment, we as a group already had made plans of what to do next. We believe that staying ahead a bit will prove to be a smart investment of time and benefit us in future assignments. If we continue development between assignments, the workload when the actual assignment is released is diminished. Even if what we develop isn’t a part of the next assignment, it will still be useful for the final product.
+
+Although the team were flexible during this assignment, Tor Kristian and Tobias will have to make up for their absence with developing in the future, as having them do the written assignments would be unfair to the rest of the team.
+
+
+We should also improve at making sure everyone is working and committing their work to gitHub as it might look like someone is not contributing or contributing less to the project. It's a group project, and everyone should contribute somewhat equally. 
+
```
Comment (MarcusAadnekvam) [inline deliverables/ObligatoriskOppgave2.md:58]: Regarding the list of high level requirements from task 1
* Allthough we wanted to start working on the visual part of the game and get something that is playable, we quickly realized that we need to create a good back-end part of the game first. This is what we ended up working on in this iteration.

Regarding testing and test driven development
* In the first iteration our codebase really only consisted of what we learnt in the guide posted on MittUiB. As the game UI actually needed to run to test anything, we couldnt really write any tests.
In this iteration we have prioritized tests and followed the principles of test driven development. We currently have 24 working tests.

File: deliverables/ObligatoriskOppgave2.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+ # Task 2 - Requirements 
+
+## User Stories 
+
+We generally went light on user stories for this release as most of the implementation is back-end related and as such has very little visibility to any users.
+As a user I expect the game does not allow me to do illegal moves
+The game should handle collisions between players or a user moving outside the bounds of the game without crashing or throwing errors.
+Criteria: Once the game has initialized and we have created a set amount of players, attempting to move a robot of the grid itself should not work. Neither should players be able to move onto tiles already occupied unless they have cards that can for example push other robots away.
```
Comment (tor-kristian) [inline deliverables/ObligatoriskOppgave2.md]: We generally went light on user stories for this release as most of the implementation is back-end related and as such has very little visibility to any users.
* As a user I expect the game does not allow me to do illegal moves
  * The game should handle collisions between players or a user moving outside the bounds of the game without crashing or throwing errors.
  * **Criteria:** Once the game has initialized and we have created a set amount of players, attempting to move a robot of the grid itself should not work. Neither should players be able to move onto tiles already occupied unless they have cards that can for example push other robots away.

**Review by tor-kristian — CHANGES_REQUESTED** [review]

File: deliverables/ObligatoriskOppgave2.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+* **Criteria**: A player should control one and only one robot.
+  * A player has HP and a robot takes damage. We need to ensure that a player can never control more than one robot at a time and that damage inflicted (and subsequently the death of a robot) affects his or hers HP.
+
+
+## Testing 
+
+We have made several test classes with several test methods, to ensure that what we have developed is functioning properly. `DirectionTest` checks if every direction to the left, right and the opposite is correct. `GridTest` checks if the proper coordinates are returned
+
+One for directions, the grid, position, player, robotest
+
```
Comment (MarcusAadnekvam) [inline deliverables/ObligatoriskOppgave2.md:91]: We are continuing to work with the principles of test driven development

File: deliverables/ObligatoriskOppgave2.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  * At this moment, the backend and the frontend parts of the game live in different worlds. As we have finished a lot of the backend that we consider necessary, the goal now will be to connect the map with the backend.
+    * Get the map connected to the current back end
+    * Create a proper GUI for the game.
+    * Program cards
+    * We will be prioritizing this before working on the logic for the actual game.
+
+Forklar kort hvilke hovedkrav dere anser som en del av MVP og hvorfor. Hvis det er gjort endringer i rekkefølge utfra hva som er gitt fra kunde, hvorfor er dette gjort?
+* We believe the following hovedkrav are a part of MVP;
+  * Krav 
+    * Reasoning
```
Comment (MarcusAadnekvam) [inline deliverables/ObligatoriskOppgave2.md:111]: We can keep the priority list of requirements from the first assignment, but make them a little more specific.
* Show a game board that is connected to the current backend of the game
* Show elements like the player and flag that are connected to the current backend game

**Review by MarcusAadnekvam — CHANGES_REQUESTED** [review]

**Comment by sondr3** [comment]
I'll add the UML-document once I've merged everything together, so I'll probably do it myself in the merge itself :+1:

**Pull request merged**